#!/usr/bin/env python
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

combination_method = []
combination_method.append(gen.const("highest_layer", int_t, 1, "Takes highest in layers"))
combination_method.append(gen.const("current_layer", int_t, 0, "Takes current layer"))
combination_method_enum = gen.enum(combination_method, "Combination methods")

decay_model = []
decay_model.append(gen.const("Linear", int_t, 0, "Linear"))
decay_model.append(gen.const("Exponential", int_t, 1, "Exponential"))
decay_model.append(gen.const("Permanent", int_t, 3, "Permanent"))
decay_model_enum = gen.enum(decay_model, "Decay models")

gen.add("enabled", bool_t, 0, "TBD", True)
gen.add("voxel_size", double_t, 0, "TBD", 0.05, 0.001, 5.0)
gen.add("combination_method", int_t, 0, "Combination methods", 1, edit_method=combination_method_enum)
gen.add("mark_threshold", double_t, 0, "TBD", 0.05, 0.001, 5.0)
gen.add("update_footprint_enabled", bool_t, 0, "TBD", True)
gen.add("track_unknown_space", bool_t, 0, "TBD", True)
gen.add("decay_model", int_t, 0, "Decay models", 0, edit_method=decay_model_enum)
gen.add("voxel_decay", double_t, 0, "TBD", -1.0, -100.0, -0.001)

exit(gen.generate("spatio_temporal_voxel_layer", "spatio_temporal_voxel_layer", "SpatioTemporalVoxelLayer"))
